script/tool/lib/src/gradle_check_command.dart

```diff
@@ -228,24 +253,31 @@ apply plugin: "com.google.cloud.artifactregistry.gradle-plugin"
         r'classpath.*gradle\.plugin\.com\.google\.cloud\.artifactregistry:artifactregistry-gradle-plugin');
     final RegExp artifactRegistryPluginApplyRegex = RegExp(
         r'apply.*plugin.*com\.google\.cloud\.artifactregistry\.gradle-plugin');
+    final RegExp artifactRegistryPluginApplyRegexGP = RegExp(
+        r'id.*com\.google\.cloud\.artifactregistry\.gradle-plugin.*version.*\b\d+\.\d+\.\d+\b');
+
 
     final bool documentationPresent = gradleLines
         .any((String line) => documentationPresentRegex.hasMatch(line));
     final bool artifactRegistryDefined = gradleLines
         .any((String line) => artifactRegistryDefinitionRegex.hasMatch(line));
     final bool artifactRegistryPluginApplied = gradleLines
         .any((String line) => artifactRegistryPluginApplyRegex.hasMatch(line));
+    final bool artifactRegistryPluginAppliedGP = gradleLines
+        .any((String line) => artifactRegistryPluginApplyRegexGP.hasMatch(line));
+
+    final bool validArtifactConfiguration = documentationPresent &&
+        ((artifactRegistryDefined && artifactRegistryPluginApplied) || artifactRegistryPluginAppliedGP);
 
-    if (!(documentationPresent &&
-        artifactRegistryDefined &&
-        artifactRegistryPluginApplied)) {
+    if (!validArtifactConfiguration && !artifactRegistryPluginAppliedGP) {
+      printError('Failed Artifact Hub validation. Include the following in '
+          'example root settings.gradle:\n$exampleSettingsArtifactHubStringGP');
+    }
+    if (!validArtifactConfiguration && !(artifactRegistryDefined && artifactRegistryPluginApplied)) {
       printError('Failed Artifact Hub validation. Include the following in '
           'example root settings.gradle:\n$exampleRootSettingsArtifactHubString');
     }
-
-    return documentationPresent &&
-        artifactRegistryDefined &&
-        artifactRegistryPluginApplied;
+    return validArtifactConfiguration;
```

Comment:
If the only thing is missing is the documentation, before it would print an example of what needed to be included that included the documentation, whereas now it would fail without any actionable message.

Reply:
I just tried out only removing only the documentation, and it does include an error message. I have a check for `documentationPresent` when `validArtifactConfiguration` is deinfed, and check for `!validArtifactConfiguration` when an error should be printed. Could you further explain why it would fail? I can add a test for this?